### PR TITLE
Ci update, bugfixes

### PR DESCRIPTION
**Pull request by Matthew-Whitlock**
Improved several CMake semantics:
- Variables now prepended with FENIX_
- mpirun tests now use cmake variables to ensure forward compatibility
- Attempts to fix issues with building on systems which have mpi headers in the default include directories, causing segfaults etc.

Merging in some bug fixes for various recovery modes (ideally a separate pr, but here we are)

Swap from Travis testing to Github actions

### CODE REVIEW

**Comment by Matthew-Whitlock** [comment]
Tests are run here, until this is merged to enable testing in this repo: https://github.com/Matthew-Whitlock/Fenix/actions/runs/4725087584/jobs/8383036470

Some tests "failed" due to timeout, but that's just a bug with ULFM where MPI_Finalize hangs occasionally.

File: CMakeLists.txt
Comment (nmm0) [inline CMakeLists.txt:49]: I'm not sure these will be transitive and might break Fenix for consumers. Would using `target_include_directories` with the MPI dirs work instead? It would simplify the code and, by default, is non-system

File: README.md
Comment (nmm0) [inline README.md:27]: I think you forgot to finish this sentence

File: CMakeLists.txt
Comment (nmm0) [inline CMakeLists.txt:19]: FWIW CTest automatically creates the `BUILD_TESTING` variable https://cmake.org/cmake/help/latest/module/CTest.html

It's probably better to use that since it's a kind of a convention. Though I think it's not very well known...

File: test/subset_internal/CMakeLists.txt
Comment (nmm0) [inline test/subset_internal/CMakeLists.txt:10]: I'm glad this was removed

File: CMakeLists.txt
Comment (nmm0) [inline CMakeLists.txt:44]: Does this actually work with >= 3.25? I thought that CMake would typically downgrade itself and not use newer features when the required CMake version was below the current.

**Review by nmm0 — CHANGES_REQUESTED** [review]
Just a couple of minor changes/questions to answer. The `FENIX_TESTS` option one isn't a big deal, I'll leave it up to you whether you want to use that or CMake's `BUILD_TESTING`

File: CMakeLists.txt
Comment (Matthew-Whitlock) [reply CMakeLists.txt]: include_directories ends up just setting the default for subsequent targets' includes, so it works similarly. It is true that this doesn't help consumers which may run into the same issue, so I've updated a move to an external cmake file and we run the same fix in the fenixConfig for consumers.

Setting FENIX_PROPAGATE_INC_FIX in Fenix sets a default value, and setting it in a given consumer overrides to decide if we actually do the include fix.

File: CMakeLists.txt
Comment (Matthew-Whitlock) [reply CMakeLists.txt]: I didn't know that, I've reverted these thanks!

File: CMakeLists.txt
Comment (Matthew-Whitlock) [reply CMakeLists.txt]: I gave it a test and it seems functional. Maybe it sets up backwards compatibility with the required version's features, but since this is a new feature it works out?

**Review by nmm0 — APPROVED** [review]
Looks good to me :)

(one minor thing -- `BUILD_EXAMPLES` isn't a cmake idiom though `BUILD_TESTS` is 🤷 )

**Pull request merged**
